src/whenIdle.test.ts

```diff
@@ -0,0 +1,23 @@
+import { whenIdle } from './whenIdle';
+
+describe('whenIdle', () => {
+	it('does not fire callback straight away', () => {
+		let fired = false;
+		const cb = () => {
+			fired = true;
+		};
+		whenIdle(cb);
+		expect(fired).toBe(false);
+	});
+
+	it('fires the callback after 600ms', async () => {
+		let fired = false;
+		const cb = () => {
+			fired = true;
+		};
+		whenIdle(cb);
+		expect(fired).toBe(false);
+		await new Promise((r) => setTimeout(r, 600));
```

Comment:
Have you tried using [Jest's timer mocks](https://jestjs.io/docs/timer-mocks) for this kind of behaviour? I expect it's more deterministic.

Reply:
Those new mocks for Jest look nice. The way Jest used globals before was painful. However, in this case I'm not sure it works. I need to advance _real_ time, not a mocked `setTimeout`, because the code is - potentially - using the idle callback. I guess you could argue we could just ignore the actual idle event and force the setTimeout but maybe that's less coverage?

Reply:
Yes you're right, it doesn't look like Jest provides any way to mock `requestIdleCallback`

Reply:
Digging a bit more, [`@shopify/jest-dom-mocks`](https://github.com/Shopify/quilt/tree/main/packages/jest-dom-mocks) looks useful for features available in the browser. Currently, Coveralls is failing because [we're only testing the fallback case](https://coveralls.io/builds/44541082/source?filename=src%2FwhenIdle.ts#L17) when `requestIdleCallback` is unavailable.

Reply:
🤔 Given the only thing this abstraction really does is implement `requestIdleCallback`, do we risk a situation where we're purely testing the mock?

I guess there's an argument to say that our code is so simple anyway so the test will be the same and there's benefit in having them, if only for documentation?

Reply:
Yeah I think there's some benefit in knowing this function does what we expect, both with and without browser support.